### PR TITLE
Ac/namespaces

### PR DESCRIPTION
**Pull request by eljuguetero**
Zwave added, 
CMake fixed

### CODE REVIEW

File: matrixlabs/malos/v1/zwave_commands.proto [hunk 1]
```diff
@@ -0,0 +1,785 @@
+// Proto generated by './zwave_proto_gen'
+// Do not edit!
+
+syntax = "proto3";
+
+package matrixlabs.malos.v1.driver;
```
Comment (maciekrb) [inline matrixlabs/malos/v1/zwave_commands.proto]: This should probably be `matrixlabs.malos.v1.zwave`

**Comment by eighteyes** [comment]
I'd like to see vision.v1.core instead of vision.v1.vision.
I'd like to see a malos.v1.core file for the driver

If we're going to break and touch everything in our system, we should consider refactoring our protobufs to be more modular.

Can we break out zigbee, wakeword, gpio/servo, IR, Everloop and microphone into their own files?

There are FacialRecognition messages inside vision.proto. etc. I'm not as familiar with CV but there are clear opportunities for giving us a solid platform to grow on, instead of just adding namespaces.

**Comment by maciekrb** [comment]
The initial idea behind the namespacing, is to make the protos more usable from other languages, i.e python. The change in structure also allows for versioning (v1, v2, etc), so we open some room for other message / service message structures. 

What if we first agree on the namespacing and workout the details on message structure on subsequent PRs ?

File: matrixlabs/malos/v1/zwave_commands.proto [hunk 2]
```diff
@@ -0,0 +1,785 @@
...
+
+option csharp_namespace = "Matrixlabs.Malos.Driver.V1";
```
Comment (maciekrb) [inline matrixlabs/malos/v1/zwave_commands.proto]: `option csharp_namespace = "Matrixlabs.Malos.Zwave.V1";`

File: matrixlabs/malos/v1/zwave_commands.proto [hunk 3]
```diff
@@ -0,0 +1,785 @@
...
+option java_multiple_files = true;
+option java_outer_classname = "DriverProto";
+option java_package = "com.matrixlabs.malos.v1";
```
Comment (maciekrb) [inline matrixlabs/malos/v1/zwave_commands.proto]:
```
option java_outer_classname = "ZwaveProto";
option java_package = "com.matrixlabs.malos.v1";
```

File: matrixlabs/malos/v1/driver.proto [hunk 4]
```diff
@@ -22,6 +23,8 @@ option java_multiple_files = true;
 option java_outer_classname = "DriverProto";
 option java_package = "com.matrixlabs.malos.v1";
 
+import "zwave_commands.proto";
```
Comment (maciekrb) [inline matrixlabs/malos/v1/driver.proto]: `import "matrixlabs/vision/v1/zwave_commands.proto";`

**Comment by eighteyes** [comment]
@maciekrb I agree, it would be best to get namespacing in as a v1 and do subsequent optimizations later. If this was all internal, no biggie. My main concern is we'll be asking our users to adapt to a big refactor later, instead of doing it sooner. My philosophy is to rip off the bandaid, because we'll have to do it anyway, and it will have less impact if we do it now vs later.

**Comment by maciekrb** [comment]
@eighteyes I think it's fine as long as we understand and agree on the implications. Can you branch from here and create a PR so we can have an idea of what you have in mind ?

**Pull request closed without merging**